scylla.go

```diff
@@ -295,9 +295,22 @@ func (p *scyllaConnPicker) Pick(t token) *Conn {
 	if c := p.conns[idx]; c != nil {
 		// We have this shard's connection
 		// so let's give it to the caller.
-		return c
+		// But only if it's not loaded too much and load is well distributed.
+		if !isHeavyLoaded(c) {
+			return c
+		}
+		alternative := p.leastBusyConn()
+		if alternative == nil || alternative.AvailableStreams() * 120 > c.AvailableStreams() * 100 {
+			return c
+		} else {
+			return alternative
+		}
```

Comment:
nit: this could be refactored to function then you could just `return p.choose(c, p.leastBusyConn())`

Reply:
I see. Will do.

Reply:
done.